nodes/network/file_path.py

```diff
@@ -111,6 +116,27 @@ def process(self):
             files.append(filepath)
         self.outputs['File Path'].sv_set(files)
 
+    # iojson stuff
+
+    def storage_set_data(self, storage):
+        '''function to get data when importing from json''' 
+
+        strings_json = storage['string_storage']
+        filenames = json.loads(strings_json)['filenames']
+        directory = json.loads(strings_json)['directory']
+        
+        self.id_data.freeze(hard=True)
+        self.set_data(directory, filenames)
+        self.id_data.unfreeze(hard=True)
+
+    def storage_get_data(self, node_dict):
+        '''function to set data for exporting json''' 
+
+        local_storage = {
+            'filenames': [{"name": file_elem.name} for file_elem in self.files], 
+            'directory': self.directory
+        }
+        node_dict['string_storage'] = json.dumps(local_storage)
```

Comment:
You love doing things over complicated, why don't just do this:
```Python
node_dict['string_storage'] = local_storage
```

Reply:
> You love doing things over complicated

that's "projection" @Durman .   But i did ` = json.dumps(local_storage)`  and not ` = local_storage` , because i referenced older _working_ implementations. At some point in the past i must have had a reason to use that.